__tests__/app.test.js

```diff
@@ -329,4 +329,53 @@ describe("GET /api/users", () => {
             })
         })
     })
+})
+
+describe("GET /api/articles?topic", () => {
+    test("status 200: responds with articles of specified topic and if topic query is omitted, responds with all articles", () => {
+        return request(app)
+        .get("/api/articles?topic=cats")
+        .expect(200)
+        .then((res) => {
+            res.body.articles.forEach((article) => {
+                expect(article).toMatchObject({
+                    author: expect.any(String),
+                    title: expect.any(String),
+                    article_id: expect.any(Number),
+                    // body: expect.any(String),
+                    topic: "cats",
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    article_img_url: expect.any(String)
+                })
+            })
+        })
+    })
+    test("status 400: responds with 'Invalid Input' if the topic does not exist", () => {
```

Comment:
This topic could exist at some point so it's not invalid, this would be a 404 and a better error message would be something like "No topic matching the topic query"